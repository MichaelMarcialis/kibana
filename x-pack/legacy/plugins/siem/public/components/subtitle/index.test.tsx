/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

import '../../mock/ui_settings';
import { TestProviders } from '../../mock';
import { Subtitle } from './index';

jest.mock('../../lib/settings/use_kibana_ui_setting');

describe('Subtitle', () => {
  test('it renders', () => {
    const wrapper = shallow(
      <TestProviders>
        <Subtitle text="Test subtitle" />
      </TestProviders>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('it renders one subtitle item', () => {
    const wrapper = mount(
      <TestProviders>
        <Subtitle text="Test subtitle" />
      </TestProviders>
    );

    expect(wrapper.find('p').length).toEqual(1);
  });

  test('it renders multiple subtitle items', () => {
    const wrapper = mount(
      <TestProviders>
        <Subtitle text={['Test subtitle 1', 'Test subtitle 2']} />
      </TestProviders>
    );

    expect(wrapper.find('p').length).toEqual(2);
  });
});
