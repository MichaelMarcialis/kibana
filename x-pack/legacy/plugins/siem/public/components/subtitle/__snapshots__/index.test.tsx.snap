// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subtitle it renders 1`] = `
<Component>
  <Subtitle
    text="Test subtitle"
  />
</Component>
`;
